draft-ietf-tls-tls13.md

```diff
@@ -3488,6 +3503,9 @@ access_denied
 decode_error
 : A message could not be decoded because some field was out of the
   specified range or the length of the message was incorrect.
+  This alert is used for errors where the message does not conform
+  to the formal protocol syntax (e.g., a vector is specified as
+  having the range <2..255> but is encoded as having 0 length.)
```

Comment:
I'm not following. There are effectively two options here:
- You have a buffer which should coterminate with the production and therefore you can detect both too short and too long
- You have a buffer which contains the production and then some other stuff afterwards in which case you can only detect the buffer being too short while processing that production.

I agree that an odd-length vector with even-sized elements is a distinct kind of error. My point is that "4 fields is received with only 3 fields present"  is a general case of too short encoding.